app/src/common/shared/com/igalia/wolvic/search/suggestions/SuggestionsProvider.java

```diff
@@ -180,11 +184,60 @@ private CompletableFuture<List<SuggestionItem>> getSearchEngineSuggestions(@NonN
         return future;
     }
 
+    private CompletableFuture<List<SuggestionItem>> removeDuplicatedItems(@NonNull final List<SuggestionItem> items) {
+        CompletableFuture<List<SuggestionItem>> future = new CompletableFuture<>();
+
+        Map<String, List<Integer>> category = new HashMap<>();
+        Map<String, Integer> best = new HashMap<>();
+
+        // Filter out duplicate items based on the URL
+        for (int i = 0; i < items.size(); i++) {
+            SuggestionItem thisItem = items.get(i);
+            // Normalize URLs to remove 'http'/'https' and trailing slashes
+            String normalizedUrl = thisItem.url.replaceAll("^https?://", "").replaceAll("/$", "");
```

Comment:
Perhaps we could get advantage of the utility functions we have in the UrlUtil class. Additionally, we could try to create a java.net.URL instance from the string and uses it's API to categorize the SutestionItem instance.

Reply:
Added some logic to handle this in UrlUtils